public/utils/pipeline/ppl_aggs_auto_suggest_task.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Task } from './task';
+import { DataPublicPluginStart } from '../../../../../src/plugins/data/public';
+
+interface Input {
+  ppl: string;
+  dataSourceId: string | undefined;
+  timeFiledName?: string;
+}
+
+export class PPLAggsAutoSuggestTask extends Task<Input, Input> {
+  searchClient: DataPublicPluginStart['search'];
+
+  constructor(searchClient: DataPublicPluginStart['search']) {
+    super();
+    this.searchClient = searchClient;
+  }
+
+  async execute<T extends Input>(v: T) {
+    let ppl = v.ppl;
+
+    if (ppl) {
+      const isPPLHasAgg = this.isPPLHasAggregation(ppl);
+      // if no aggregation, will try auto suggest one
+      if (!isPPLHasAgg) {
+        if (v.timeFiledName) {
+          const dateRangePPL = `${ppl} | stats min(${v.timeFiledName}) as min, max(${v.timeFiledName}) as max`;
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          let res: any;
+          try {
+            res = await this.searchClient
+              .search(
+                { params: { body: { query: dateRangePPL } }, dataSourceId: v.dataSourceId },
+                { strategy: 'pplraw' }
+              )
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              .toPromise<any>();
+
+            if (res.rawResponse.total === 0) {
+              ppl = `${ppl} | stats count()`;
+            } else {
+              // get min and max and calculate proper interval by range
+              const min = res.rawResponse.jsonData[0].min;
```

Comment:
If the generated ppl contains time range condition, then should we use that condition directly to narrow down the time range? Too many buckets maybe created if we use the max/min time in the index, which has poor performance.

Reply:
we didn't remove any filter in original ppl, what we do is get the range for existing query result and append a date histogram aggregation